Dockerfile

```diff
@@ -1,13 +1,22 @@
+# Stage 1: Build node portion of the H app.
 FROM node:alpine as build
 
 ENV NODE_ENV production
 
-COPY . .
+# Build node dependencies.
+COPY package-lock.json ./
+COPY package.json ./
```

Comment:
These `COPY` commands could be folded into a single command with `COPY src1 src2 ... dest` since these files usually change together. However I don't have numbers for how much value there is in doing that. https://stackoverflow.com/questions/47079114/should-i-minimize-the-number-of-docker-layers has some related discussion.